sickbeard/providers/bithdtv.py

```diff
@@ -0,0 +1,161 @@
+# coding=utf-8
+# Author: p0psicles
+#
+# This file is part of Medusa.
+#
+# Medusa is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Medusa is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Medusa. If not, see <http://www.gnu.org/licenses/>.
+
+import re
+from requests.utils import dict_from_cookiejar
+
+from sickbeard import logger, tvcache
+from sickbeard.bs4_parser import BS4Parser
+
+from sickrage.helper.common import convert_size, try_int
+from sickrage.providers.torrent.TorrentProvider import TorrentProvider
+
+
+class BithdtvProvider(TorrentProvider):  # pylint: disable=too-many-instance-attributes
+
+    def __init__(self):
+
+        # Provider Init
+        TorrentProvider.__init__(self, "BITHDTV")
+
+        # Credentials
+        self.username = None
+        self.password = None
+
+        # Torrent Stats
+        self.minseed = 0
+        self.minleech = 0
+        self.freeleech = True
+
+        # URLs
+        self.url = 'https://www.bit-hdtv.com/'
+        self.urls = {
+            'login': self.url + 'takelogin.php',
+            'search': self.url + 'torrents.php',
+        }
+        # Proper Strings
+
+        # Cache
+        self.cache = tvcache.TVCache(self, min_time=10)  # Only poll BitHDTV every 10 minutes max
+
+    def login(self):
+        if any(dict_from_cookiejar(self.session.cookies).values()):
+            return True
+
+        login_params = {
+            'username': self.username.encode('utf-8'),
+            'password': self.password.encode('utf-8'),
+        }
+
+        response = self.get_url(self.urls['login'], post_data=login_params, returns='text')
+        if not response:
+            logger.log(u"Unable to connect to provider", logger.WARNING)
+            self.session.cookies.clear()
+            return False
+
+        if '<h2>Login failed!</h2>' in response:
+            logger.log(u"Invalid username or password. Check your settings", logger.WARNING)
+            self.session.cookies.clear()
+            return False
+
+        return True
+
+    def search(self, search_strings, age=0, ep_obj=None):  # pylint: disable=too-many-locals, too-many-branches
+        results = []
+        if not self.login():
+            return results
+
+        # Search Params
+        search_params = {'cat': '12'} if 'Season' in search_strings else {'cat': '10'}
```

Comment:
Yes it does. It can have rss, episode and season when A season Pack search is started. I need to send An alternative Cat number. 12, for when searching for season Pack. Dont know how other providers do this?